metals/src/main/scala/scala/meta/internal/metals/MtagsResolver.scala

```diff
@@ -49,7 +66,16 @@ object MtagsResolver {
 
       def fetch(tries: Int = 0): State = {
         try {
-          val jars = Embedded.downloadMtags(scalaVersion)
+          val metalsVersion = removedScalaVersions.getOrElse(
+            scalaVersion,
+            BuildInfo.metalsVersion,
+          )
+          if (metalsVersion != BuildInfo.metalsVersion) {
+            scribe.warn(
+              s"$scalaVersion is no longer supported in the current Metals versions, using the last known supported version $metalsVersion"
+            )
+          }
+          val jars = Embedded.downloadMtags(scalaVersion, metalsVersion)
```

Comment:
so this tries to fetch mtags for current or some previous version and fetching for previous version is safe because mtags are supposed to be binary compatible, yes?

Reply:
> we can always point to the last supported version and allow users to use it at their own risk.

Hmm, is `use it at their own risk` just `scribe.warn`? I think many people doesn't open metals output/metals.log unless they're reporting bug

Reply:
It is binary compatible :D And I added a test that checks that, but yeah. If we add a new method it will have a default so it will just not work for the older versions.

Reply:
> Hmm, is use it at their own risk just scribe.warn? I think many people doesn't open metals output/metals.log unless they're reporting bug

They should also get the warning in the message request. This is only if we need to debug something, so when someone reports a bug

Reply:
Thanks for clarification